### DISCUSSION TITLE
Using a POST response body

### DISCUSSION THREAD

**Discussion by flpms** [Budibase/budibase#11976] [Help] [answered]
Not every API it's a perfect world like described in the docs. I have a POST request that bring data, it's kind of a query request, so I need to display this data, instead make a GET request.

I tried follow what it's described [here in the docs](https://docs.budibase.com/docs/rest-queries) for queries without success. Even worse the request it's doubled one correct and another one with 400 due an empty body.

So the question is, how can I display those results for my user in budibase?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @flpms,

You should be able to save the results received from a post API request to [state](https://docs.budibase.com/docs/app-state) using the [actions](https://docs.budibase.com/docs/actions) bindings, in this binding you should have access to the result of the post request.
<img width="327" alt="Screenshot 2023-10-09 at 09 31 42" src="https://github.com/Budibase/budibase/assets/126772285/88dc206d-d517-4e47-9d1e-a81265e64091">

Drawer were the query result binding goes.
<img width="891" alt="Screenshot 2023-10-09 at 09 31 51" src="https://github.com/Budibase/budibase/assets/126772285/4704de7b-cab3-4cc5-adc7-37d9bd85d5a9">

I hope this helps!
